libwayshot/src/dispatch.rs

```diff
@@ -184,30 +183,27 @@ impl Dispatch<ZwlrScreencopyFrameV1, ()> for CaptureFrameState {
                     exit(1);
                 }
             }
-            zwlr_screencopy_frame_v1::Event::Flags { .. } => {
-                tracing::debug!("Received Flags event");
-            }
             zwlr_screencopy_frame_v1::Event::Ready { .. } => {
                 // If the frame is successfully copied, a “flags” and a “ready” events are sent. Otherwise, a “failed” event is sent.
                 // This is useful when we call .copy on the frame object.
-                tracing::debug!("Received Ready event");
+                tracing::trace!("Received Ready event");
                 frame.state.replace(FrameState::Finished);
             }
             zwlr_screencopy_frame_v1::Event::Failed => {
-                tracing::debug!("Received Failed event");
+                tracing::trace!("Received Failed event");
                 frame.state.replace(FrameState::Failed);
             }
             zwlr_screencopy_frame_v1::Event::Damage { .. } => {
-                tracing::debug!("Received Damage event");
+                tracing::trace!("Received Damage event");
             }
             zwlr_screencopy_frame_v1::Event::LinuxDmabuf { .. } => {
-                tracing::debug!("Received LinuxDmaBuf event");
+                tracing::trace!("Received LinuxDmaBuf event");
             }
             zwlr_screencopy_frame_v1::Event::BufferDone => {
-                tracing::debug!("Received bufferdone event");
+                tracing::trace!("Received bufferdone event");
                 frame.buffer_done.store(true, Ordering::SeqCst);
             }
-            _ => unreachable!(),
```

Comment:
What I'd want to avoid however, is when there is a newer version of screencopy with a new event, that we'd reach this leading to an error. If the event enum is exhaustive, we can remove the `_ => {}` and be sure this won't happen anyhow. If it's not, I'd want to keep `_ => {}` and not `_ => unreachable!()` as it might lead to errors in the future, no?

Reply:
I understand where you're coming from. I believe we request the exact version of screencopy we want from the queuehandle anyways, when we update the version of scrcpy, the compiler *should* warn us of an unhandled cases right as it's an exhaustive enum? So maybe we can just expand the handler with some polyfill (eg: //TODO: handle this later/ignored event) for the cases we don't currently handle?

Reply:
> I believe we request the exact version of screencopy we want from the queuehandle anyways

Now that you mention this, we need to update the code afaik to say version X or higher as I think right now if a new version is introduced, we won't support a compositor that reports a higher version even though it should also support the older version. 🤔 Though unrelated to this PR.


> when we update the version of scrcpy, the compiler should warn us of an unhandled cases right as it's an exhaustive enum?

It might warn us (though don't think so because we have `_ => {}`), a newer version will still hit `_ => unreachable!()`. We simply cannot guarantee this is unreachable with a non exhaustive enum.